### PR TITLE
Requirejs migration: locations

### PR DESCRIPTION
**Pull request by orangejenny**
@gcapalbo / @esoergel 

[QA ticket](https://manage.dimagi.com/default.asp?278133)

### CODE REVIEW

File: corehq/apps/locations/static/locations/js/location_drilldown.js [hunk 1]
```diff
@@ -92,7 +91,7 @@ hqDefine('locations/js/location_drilldown', [
 
         // load location hierarchy and set initial path
         model.load = function(locs, selected) {
-            model.root(new model.func({name: '_root', children: locs, auto_drill: model.auto_drill}, model));
+            model.root(new model.func({name: '_root', children: locs, auto_drill: model.auto_drill}, model, options.required));
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/locations/static/locations/js/location_drilldown.js]: Ignoring these, as they aren't straightforward to switch to a functional style.

File: custom/intrahealth/templates/intrahealth/location_filter.html [hunk 2]
```diff
@@ -29,7 +27,8 @@
             "auto_drill": null,
             "loc_filter": false,
             "loc_url": location_url,
-            "func": IntrahealthLocModel
+            "func": IntrahealthLocModel,
+            "required": {{ required }},
```
Comment (esoergel) [inline custom/intrahealth/templates/intrahealth/location_filter.html]: :+1:

**Review by esoergel — APPROVED** [review]
Fantastic PR!

**Review by esoergel — APPROVED** [review]

**Comment by nickpell** [comment]
Getting a merge conflict when rebuilding staging - mind taking a look?

**Comment by orangejenny** [comment]
@nickpell updated

**Comment by nickpell** [comment]
thanks!

**Pull request merged**